src/Database/PostgreSQL/Consumers/Config.hs

```diff
@@ -21,12 +22,16 @@ data Action
   | RerunAfter Interval
   | RerunAt UTCTime
   | Remove
-    deriving (Eq, Ord, Show)
+  deriving (Eq, Ord, Show)
 
 -- | Result of processing a job.
 data Result = Ok Action | Failed Action
   deriving (Eq, Ord, Show)
 
+-- | The mode the consumer will run in.
+data Mode = Standard | Duplicating (RawSQL ())
```

Comment:
Should we allow deduplicating on the primary key row of the jobs table like `ccMode = Duplicating "id"`? If you do this right now you get an ambiguity error in the sql query used for reserving jobs in the consumer because part of the query used in `reserveJobs` becomes `SELECT id, id ...`

Reply:
My approach would be **not** to do that if it's not necessary for proper function now. It can be added later if there's a need for it. And maybe document it somewhere that you can't deduplicate based on fields that are called `id`.